pkg/master/master.go

```diff
@@ -306,7 +306,17 @@ func (m *Master) InstallAPIs(c *Config) {
 			glog.Warningf("Problem initializing API group %q, skipping.", group)
 			continue
 		}
-		glog.V(1).Infof("Enabling API group %q.", group)
+		// We need to filter apiGroupInfo.GroupMeta.GroupVersions so that it reflects
+		// what's actually enabled. By default this value contains all versions
+		// registered for a group.
+		groupVersions := []unversioned.GroupVersion{}
+		for _, gv := range apiGroupInfo.GroupMeta.GroupVersions {
+			if c.APIResourceConfigSource.AnyResourcesForVersionEnabled(gv) {
+				glog.V(1).Infof("Enabling API group %q.", gv)
+				groupVersions = append(groupVersions, gv)
+			}
+		}
+		apiGroupInfo.GroupMeta.GroupVersions = groupVersions
```

Comment:
Agreed, mutating this structure doesn't make sense, since its description clearly state it's [all available versions](https://github.com/kubernetes/kubernetes/blob/520cd7e6f386980d062505d6028f102037304a7a/pkg/apimachinery/types.go#L30).